packages/truffle-external-compile/index.js

```diff
@@ -39,6 +41,21 @@ const runCommand = promisify(function (command, options, callback) {
   });
 });
 
+function decodeContents (contents) {
+  // HACK to test if contents has binary data
+  // from https://stackoverflow.com/a/49773659
+  if (/\ufffd/.test(contents.toString())) {
```

Comment:
Hm, so I was thinking that `contents.toString()` was "correctly" rendering those unreadable characters. Is that not the case in all situations?

Do you have any other ideas as to how to detect this, @axic? I'm trying to avoid making these files parsed in a property-specific way, and also trying to avoid requiring the user to specify the encoding on a per-property basis.

Reply:
I would suggest that if the file starts with `0x` (and passes the hex test for the rest, e.g. has only hex characters) then read it as a hex string otherwise consider it a binary.

Reply:
Mmm. OK so how about:

- If it parses as JSON, it's JSON
- If it starts with `0x`, it's hex
- Otherwise, it's binary